datafusion/sql/src/planner.rs

```diff
@@ -563,3 +570,21 @@ pub fn object_name_to_qualifier(
         .collect::<Vec<_>>()
         .join(" AND ")
 }
+#[test]
+fn test_substitute_tilde_with_home() {
+    std::env::set_var("HOME", "/home/user");
+    let input = "~/Code/arrow-datafusion/benchmarks/data/tpch_sf1/part/part-0.parquet";
+    let expected =
+        "/home/user/Code/arrow-datafusion/benchmarks/data/tpch_sf1/part/part-0.parquet";
+    let actual = substitute_tilde(input.to_string());
+    assert_eq!(actual, expected);
+    std::env::remove_var("HOME");
```

Comment:
This test will leave the `HOME` variable unset, even if it set at the beginning of the test.